### PR TITLE
[spi-hdlc-adapter] add syslog to record signal

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c [hunk 1]
```diff
@@ -171,6 +171,18 @@ static int sSpiSmallPacketSize  = 32; // in bytes
...
+{
+    SIGNAL_INT,
+    SIGNAL_TERM,
+    SIGNAL_HUP,
+    SIGNAL_INVALID,
+} SignalType;
+
+static const char sSignalMessages[][20] = {"\nCaught SIGINT!\n", "\nCaught SIGTERM!\n", "\nCaught SIGHUP!\n"};
+
+static SignalType sCaughtSignal = SIGNAL_INVALID;
```
Comment (bukepo) [inline tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: is it possible to define as int?

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c [hunk 2]
```diff
@@ -171,6 +171,18 @@ static int sSpiSmallPacketSize  = 32; // in bytes
...
 
+typedef enum SignalType
+{
+    SIGNAL_INT,
+    SIGNAL_TERM,
+    SIGNAL_HUP,
+    SIGNAL_INVALID,
+} SignalType;
+
+static const char sSignalMessages[][20] = {"\nCaught SIGINT!\n", "\nCaught SIGTERM!\n", "\nCaught SIGHUP!\n"};
```
Comment (bukepo) [inline tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: use https://linux.die.net/man/3/strsignal?

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c [hunk 3]
```diff
@@ -205,13 +217,12 @@ static uint64_t sHdlcRxBadCrcCount         = 0;
...
 {
-    static const char message[] = "\nCaught SIGINT!\n";
-
     sRet = EXIT_QUIT;
 
     // Can't use syslog() because it isn't async signal safe.
     // So we write to stderr
-    IGNORE_RETURN_VALUE(write(STDERR_FILENO, message, sizeof(message) - 1));
+    IGNORE_RETURN_VALUE(write(STDERR_FILENO, sSignalMessages[SIGNAL_INT], strlen(sSignalMessages[SIGNAL_INT])));
+    sCaughtSignal = SIGNAL_INT;
```
Comment (bukepo) [inline tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: `sCaughtSignal = sig`?

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c (hunk 1 above)
Comment (zhanglongxia) [reply tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: Updated.

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c (hunk 3 above)
Comment (zhanglongxia) [reply tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: Updated.

File: tools/spi-hdlc-adapter/spi-hdlc-adapter.c (hunk 2 above)
Comment (zhanglongxia) [reply tools/spi-hdlc-adapter/spi-hdlc-adapter.c]: Good idea.

**Pull request merged**